enhancements/machine-config/onprem-namespaces.md

```diff
@@ -0,0 +1,134 @@
+---
+title: namespaces-for-on-prem-networking-services
+authors:
+  - "@cybertron"
+reviewers:
+  - "@smarterclayton"
+  - "@celebdor"
+  - "@mandre"
+  - "@rgolangh"
+  - "@patrickdillon"
+  - "@kikisdeliveryservice"
+approvers:
+  - Someone from MCO
+  - Someone from architecture
+creation-date: 2020-07-24
+last-updated: 2020-07-27
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Namespaces for On-Prem Networking Services
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+When support for on-premise deployments on baremetal, OpenStack, OVirt, and
+VSphere was added, some additional networking services were needed to provide
+functionality that would come from cloud services on other platforms. These
+services were added in namespaces called openshift-kni-infra,
+openshift-openstack-infra, etc. There are two problems with this: 1) The
+namespaces are created in all deployments, including on platforms other
+than the one that uses the namespace and 2) The openshift-x-infra name scheme
+does not match the existing pattern for OpenShift namespaces.
```

Comment:
> existing pattern for OpenShift namespaces

Do we have link to these existing pattern?

Reply:
Just what was written in https://bugzilla.redhat.com/show_bug.cgi?id=1815561 AFAIK:

"In general, namespaces must define a component (infra is not a component) and follow an existing pattern (-operator for operators, * for operand).  Also, optional component namespaces should not be created for platforms that are not installed.  If the component is a singleton per platform, then they should reuse the same namespace (since a cluster can have only one platform)."